tests/console/osc.pm

```diff
@@ -0,0 +1,81 @@
+# SUSE's openQA tests
+#
+# Copyright © 2021 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+# Author: Gustavo Figueira <gfigueira@suse.com>
+
+use base "consoletest";
+use testapi;
+use utils;
+use Utils::Backends 'use_ssh_serial_console';
+use version_utils qw(is_sle get_os_release is_tumbleweed);
+use strict;
+use warnings;
+
+sub repos_and_pkgs_sle {
+    my ($hub_version, $arch) = @_;
+    my $repo_version = $hub_version;
+    $repo_version =~ s/\./_SP/;
+    assert_script_run("SUSEConnect --product PackageHub/" . $hub_version . "/" . $arch);
+    zypper_call("ar http://download.suse.de/ibs/SUSE:/CA/SLE_" . $repo_version . "/ suseca");
+    zypper_call("in osc obs-service-source_validator ca-certificates-suse");
+}
+
+sub prepare_oscrc {
+    my ($oscrc_path, $IBSUSER, $SECRET) = @_;
+    assert_script_run("wget --quiet " . data_url('oscrc') . " -O " . $oscrc_path, timeout => 30);
+    assert_script_run("perl -pi -e 's/TAGIBSUSER/" . $IBSUSER . "/g' " . $oscrc_path);
+    assert_script_run("perl -pi -e 's/TAGSECRET/" . $SECRET . "/g' " . $oscrc_path);
+}
+
+sub co_and_build {
+    my ($codestream, $testpkg) = @_;
+    my $api_opensuse = "https://api.opensuse.org";
+
+    if (is_sle) {
+        assert_script_run("yes 1 | osc co " . $codestream . " " . $testpkg);
+    } else {
+        assert_script_run("yes 1 | osc -A " . $api_opensuse . " co " .  $codestream . " " . $testpkg);
+    }
+    assert_script_run("cd " . $codestream . "/" . $testpkg . "; yes 1 | osc build", timeout => 300);
+}
+
+sub run {
+    check_var("BACKEND", "ipmi") ? use_ssh_serial_console : select_console 'root-console';
+
+    my $IBSUSER = get_var("IBSUSER", "");
+    my $SECRET = get_var("SECRET", "");
+    my $arch = get_var("ARCH");
+    my ($sle_major_version, $sle_sp) = get_os_release;
+    my $testpkg = "perl-IO-String";
+    my $oscrc_path = "/root/.oscrc";
+    my $hub_version = $sle_major_version . "." . $sle_sp;
+    my $sle_cs = "SUSE:SLE-" . $sle_major_version . ":Update";
+
+    # 1) Repositories and packages needed
+    if (is_sle) {
+        repos_and_pkgs_sle($hub_version, $arch);
+    } else {
+        zypper_call("in osc");
+    }
+
+    # 2) Setting osc config file
+    prepare_oscrc($oscrc_path, $IBSUSER, $SECRET);
+
+    # 3) Checking-out and building locally
+    if (is_sle) {
+        co_and_build($sle_cs, $testpkg);
+    } else {
+        co_and_build("openSUSE:Factory", $testpkg);
+    }
+}
+
+sub test_flags {
+    return get_var('PUBLIC_CLOUD') ? {milestone => 0, fatal => 1, no_rollback => 1} : {milestone => 1, fatal => 1};
```

Comment:
```suggestion
    {milestone => 0, fatal => 1, no_rollback => get_var('PUBLIC_CLOUD')};
```

should work. Maybe you need `get_var('PUBLIC_CLOUD', 0)`

But also I wonder why make that distinction here at all? Wouldn't this apply to *all* test modules within public cloud? Then it should not be duplicated everywhere.

Unfortunately I already see it duplicated in multiple modules (`git grep get_var.*PUBLIC_CLOUD.*rollback'`)

Reply:
@ge0r -> I need your pointers here

Reply:
@pdostal, might need your input here

Reply:
Wow, I am very excited of this test module! 🚀 

No `PUBLIC_CLOUD` check is needed tho:
```suggestion
    return {milestone => 1, fatal => 1};
```